app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Adapters/FavoritesListAdapter.java

```diff
@@ -70,46 +72,53 @@ public ViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
     }
 
     public int getItemCount() {
-        return this.mFavList.length;
+        return this.mFavList.size();
     }
 
     public void onBindViewHolder(@Nullable ViewHolder holder, int position) {
         if (holder == null) {
             throw new TypeCastException("null cannot be cast to non-null type com.cornellappdev.android.eatery.FavoritesFavoritesListAdapter.TextAdapterViewHolder");
         } else {
             FavoritesListAdapter.TextAdapterViewHolder holder2 = (FavoritesListAdapter.TextAdapterViewHolder)holder;
-            holder2.getFavoriteName().setText(mFavList[position].getStartPlace().getName() + "--" +
-                    mFavList[position].getEndPlace().getName());
-
-            Map<String, String> mapString = new HashMap<String, String>();
-            mapString.put("start", mFavList[position].getStartPlace().toString());
-            mapString.put("end", mFavList[position].getEndPlace().toString());
-            mapString.put("arriveBy", String.valueOf(false));
-            mapString.put("destinationName", mFavList[position].getEndPlace().getName());
+            holder2.getFavoriteName().setText(mFavList.get(position).getStartPlace().getName() + "--" +
+                    mFavList.get(position).getEndPlace().getName());
 
             Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("\"America/NewYork\""));
             int secondsEpoch = (int) (calendar.getTimeInMillis()/1000L);
-            mapString.put("time", String.valueOf(secondsEpoch));
-
-            Endpoint searchEndpoint = new Endpoint()
-                    .queryItems(mapString)
-                    .path("route")
-                    .method(Endpoint.Method.GET);
-
-            FutureNovaRequest.make(Route[].class, searchEndpoint).thenAccept((APIResponse<Route[]> response) -> {
 
-                ArrayList<Route> routes = new ArrayList<Route>();
-                for(Route r:  response.getData()){
-                    routes.add(r);
-                }
-                mAllRoutesToFavorites.put(position, routes);
-                mOptimalRoutes[position] = routes.get(0);
+            // Getting the other route options
+            Map<String, String> map = new HashMap<String, String>();
```

Comment:
pretty sure `Map<String, String> map = new HashMap<>();` will work here as well, is there any reason you're not making it a `HashMap` instead?